### PR TITLE
Resolve most linting errors

### PR DESCRIPTION
**Pull request by cag**
All the remaining errors are [`jsx-a11y/label-has-for`](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-for.md) errors.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -9,11 +9,20 @@
     "react/forbid-prop-types": [1, { "forbid": ['object', 'any'] }],
     "class-methods-use-this" : 0,
     "semi": ["error", "never"],
+    "max-len": ['error', 120, 2, {
+      ignoreUrls: true,
+      ignoreComments: false,
+      ignoreRegExpLiterals: true,
+      ignoreStrings: true,
+      ignoreTemplateLiterals: true,
```
Comment (andre-meyer) [inline .eslintrc]: Hmm, very long strings and template literals should still cause linter errors imo.

File: src/components/DecimalValue/index.js [hunk 2]
```diff
@@ -23,4 +24,9 @@ const DecimalValue = ({ value, decimals = 4 }) => {
   return <span>{text}</span>
 }
 
+DecimalValue.propTypes = {
+  value: PropTypes.string,
```
Comment (andre-meyer) [inline src/components/DecimalValue/index.js]: Actually also might be a Decimal-type Object

File: src/components/FormCheckbox/index.js [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
       <input id={`formCheckbox_${input.name}`} className={`checkboxField__input ${bemifyClassName(className, 'input')}`} type="checkbox" {...input} />
       <span className={`checkboxField__textWrapper ${bemifyClassName(className, 'textWrapper')}`}>{ text }</span>
     </label>
   </div>
-  )
+)
+
+Checkbox.propTypes = {
+  input: PropTypes.object,
+  label: PropTypes.string,
```
Comment (andre-meyer) [inline src/components/FormCheckbox/index.js]: Please use { fieldPropTypes } from redux-form here (spread): https://github.com/erikras/redux-form/blob/master/src/propTypes.js#L84

File: src/components/FormDateTimePicker/index.js [hunk 4]
```diff
@@ -20,4 +21,10 @@ const FormDateTimePicker = ({ label, input, validDateCheck }) => {
   )
 }
 
+FormDateTimePicker.propTypes = {
+  label: PropTypes.string,
+  input: PropTypes.object,
```
Comment (andre-meyer) [inline src/components/FormDateTimePicker/index.js]: Same here: https://github.com/erikras/redux-form/blob/master/src/propTypes.js#L84

File: src/components/FormEditableList/index.js [hunk 5]
```diff
@@ -96,4 +95,9 @@ class FormEditableList extends Component {
   }
 }
 
+FormEditableList.propTypes = {
+  input: PropTypes.object,
```
Comment (andre-meyer) [inline src/components/FormEditableList/index.js]: Same here: https://github.com/erikras/redux-form/blob/master/src/propTypes.js#L84

File: src/components/FormInput/index.js [hunk 6]
```diff
@@ -15,6 +16,17 @@ const Input = ({ input, meta: { error }, label, type, className, placeholder })
...
+
+Input.propTypes = {
+  input: PropTypes.object,
+  label: PropTypes.string,
+  type: PropTypes.string,
+  className: PropTypes.string,
+  placeholder: PropTypes.string,
+  meta: PropTypes.shape({
+    error: PropTypes.string,
+  }),
```
Comment (andre-meyer) [inline src/components/FormInput/index.js]: And here: https://github.com/erikras/redux-form/blob/master/src/propTypes.js#L84

File: src/components/FormRadioButton/index.js [hunk 7]
```diff
@@ -21,10 +22,23 @@ const FormRadioButton = ({ input, radioValue, text, className, highlightColor })
...
 )
 
+FormRadioButton.propTypes = {
+  input: PropTypes.object,
+  radioValue: PropTypes.string,
+  text: PropTypes.string,
+  className: PropTypes.string,
+  highlightColor: PropTypes.string,
+}
+
```
Comment (andre-meyer) [inline src/components/FormRadioButton/index.js]: And here: https://github.com/erikras/redux-form/blob/master/src/propTypes.js#L84

File: src/components/FormSelect/index.js [hunk 8]
```diff
@@ -1,20 +1,38 @@
...
+        ))}
+      </select>
+    </div>
+  )
+}
+
+const valueType = PropTypes.string
+
+FormSelect.propTypes = {
+  input: PropTypes.object,
```
Comment (andre-meyer) [inline src/components/FormSelect/index.js]: https://github.com/erikras/redux-form/blob/master/src/propTypes.js#L84

File: src/components/FormSlider/index.js [hunk 9]
```diff
@@ -1,14 +1,15 @@
 import React from 'react'
-import { isNumber } from 'lodash'
+import PropTypes from 'prop-types'
 
 import './formSlider.less'
 
-const Slider = ({ input, input: { value }, label, type, min, max, decimals, unit }) => {
-  let displayValue = parseFloat(value)
+const Slider = ({ input, label, min, max, decimals, unit }) => {
```
Comment (andre-meyer) [inline src/components/FormSlider/index.js]: Was this a linter error or just weird to read?

File: .eslintrc (hunk 1 above)
Comment (cag) [inline .eslintrc]: Perhaps. I basically copied this from the airbnb preset and changed the number to 120. Making the rule look only at the line number introduces more linter errors.

File: src/components/DecimalValue/index.js (hunk 2 above)
Comment (cag) [inline src/components/DecimalValue/index.js]: Ah, okay. Could it also be a BigNumber?

File: src/components/FormSlider/index.js (hunk 9 above)
Comment (cag) [inline src/components/FormSlider/index.js]: Oops, I just feel it was weird. Probably should have committed it separate.

File: src/components/FormCheckbox/index.js (hunk 3 above)
Comment (cag) [inline src/components/FormCheckbox/index.js]: Okay, I will.

**Comment by cag** [comment]
I'll submit corrections when I get a chance to sit down.

**Comment by cag** [comment]
Alright, I left the very long strings and template literals linter rule exception in there because it was there before and taking them out introduces a bunch of errors.

I also left the FormSlider edit in there. If you don't like it, I can revert that change.

**Pull request merged**